src/app/credExplorer/WeightConfig.js

```diff
@@ -84,55 +104,53 @@ export class WeightConfig extends React.Component<Props, State> {
     );
   }
 
-  componentDidMount() {
-    this.fire();
-  }
-
   fire() {
     const {edgeWeights, nodeWeights} = this.state;
-    const edgePrefixes = Array.from(edgeWeights.entries()).map(
-      ([prefix, {logWeight, directionality}]) => ({
-        prefix,
-        weight: 2 ** logWeight,
-        directionality,
-      })
-    );
-    const nodePrefixes = Array.from(nodeWeights.entries()).map(
-      ([prefix, logWeight]) => ({prefix, weight: 2 ** logWeight})
-    );
+    LocalStore.set(EDGE_WEIGHTS_KEY, edgeWeights);
+    LocalStore.set(NODE_WEIGHTS_KEY, nodeWeights);
+    const edgePrefixes = Object.keys(edgeWeights).map((key) => {
+      const {logWeight, directionality} = edgeWeights[key];
+      const prefix: EdgeAddressT = (key: any);
+      return {prefix, weight: 2 ** logWeight, directionality};
+    });
+    const nodePrefixes = Object.keys(nodeWeights).map((key) => ({
+      prefix: ((key: any): NodeAddressT),
+      weight: 2 ** nodeWeights[key],
+    }));
     const edgeEvaluator = byNodeType(byEdgeType(edgePrefixes), nodePrefixes);
     this.props.onChange(edgeEvaluator);
   }
 }
 
 class EdgeConfig extends React.Component<{
-  edgeWeights: Map<EdgeAddressT, UserEdgeWeight>,
-  onChange: (Map<EdgeAddressT, UserEdgeWeight>) => void,
+  edgeWeights: EdgeWeights,
+  onChange: (EdgeWeights) => void,
 }> {
   render() {
     const controls = [];
-    for (const [key, currentValue] of this.props.edgeWeights.entries()) {
+    for (const key of Object.keys(this.props.edgeWeights)) {
+      const {logWeight} = this.props.edgeWeights[key];
       controls.push(
         <label style={{display: "block"}} key={key}>
           <input
             type="range"
             min={-10}
             max={10}
             step={0.1}
-            value={currentValue.logWeight}
+            value={logWeight}
             onChange={(e) => {
               const value: number = e.target.valueAsNumber;
-              const edgeWeights = new Map(this.props.edgeWeights);
-              const oldValue = edgeWeights.get(key);
+              const edgeWeights = {...this.props.edgeWeights};
```

Comment:
Objects have unchecked property access, so this can be elided to:

```js
const value: number = e.target.valueAsNumber;
const edgeWeights = {
  ...this.props.edgeWeights,
  [key]: {...this.props.edgeWeights[key], logWeight: value},
};
this.props.onChange(edgeWeights);
```

Reply:
Done.